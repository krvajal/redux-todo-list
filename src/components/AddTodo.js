import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { addTodoAction } from "../actionCreators";

const AddTodo = ({ dispatch }) => {
  let input;
  return (
    <header className="header">
      <h1>TODOS</h1>
      <form
        onSubmit={evt => {
          evt.preventDefault();
          dispatch(addTodoAction(input.value));
          input.value = "";
        }}
      >
        <input className="new-todo" ref={node => (input = node)} />
      </form>
    </header>
  );
};

export default connect()(AddTodo);
