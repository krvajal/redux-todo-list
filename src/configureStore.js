import { createStore, applyMiddleware } from "redux";
import { loadState, saveState } from "./localStorage";
import throttle from "lodash/throttle";
import todoApp from "./reducers/index";

const logger = store => next => {
  if (!console.group) {
    // not available in some browsers
    return next;
  }
  return action => {
    console.group(action.type);
    console.log("%c prev state", "color: gray", store.getState());
    console.log("%c action", "color: blue", action);
    const returnValue = next(action);
    console.log("%c next state", "color: green", store.getState());
    console.groupEnd(action.type);
    return returnValue;
  };
};

const promise = store => next => action => {
  // chek if the action is a promise though
  if (typeof action.then === "function") {
    // the action should return an action object
    return action.then(next);
  }
  return next(action);
};

// const wrapDispatchWithMiddlewares = (store, middlewares) => {
//   middlewares
//     .slice()
//     .reverse()
//     .forEach(middleware => {
//       store.dispatch = middleware(store)(store.dispatch);
//     });
// };
const configureStore = () => {
  const persistedState = loadState();

  // the order of the middlewares are the order
  // in which the actions progress throught them
  const middlewares = [promise];
  if (process.env.NODE_ENV !== "production") {
    middlewares.push(logger);
  }
  // wrapDispatchWithMiddlewares(store, middlewares);

  const store = createStore(
    todoApp,
    persistedState, //window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(...middlewares)
  );

  store.subscribe(
    throttle(() => {
      saveState({
        todos: store.getState().todos
      });
    }, 1000)
  );
  return store;
};

export default configureStore;
