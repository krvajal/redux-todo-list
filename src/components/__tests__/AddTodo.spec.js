import React from "react";
import { shallow } from "enzyme";
import AddTodo from "../AddTodo";
import configureStore from "../../configureStore";

test("Add todo renders", () => {
  const component = shallow(<AddTodo store={configureStore()} />);
  expect(component).toMatchSnapshot();
});
