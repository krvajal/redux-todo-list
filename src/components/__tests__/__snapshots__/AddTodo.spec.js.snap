// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add todo renders 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="new-todo"
    placeholder="What needs to be done?"
  />
</form>
`;
